packages/xo-server-sdn-controller/src/index.js

```diff
@@ -327,12 +327,30 @@ class SDNController extends EventEmitter {
 
   // ===========================================================================
 
-  async _createPrivateNetwork({
+  _createPrivateNetwork({
     xoPool,
     networkName,
     networkDescription,
     encapsulation,
     xoPif,
+  }) {
+    return this._createPrivateNetworkWithVni({
+      xoPool,
+      networkName,
+      networkDescription,
+      encapsulation,
+      xoPif,
+      vni: ++this._prevVni,
+    })
+  }
+
+  async _createPrivateNetworkWithVni({
+    xoPool,
+    networkName,
+    networkDescription,
+    encapsulation,
+    xoPif,
+    vni,
```

Comment:
You don't need to create two methods for this, simply inject `vni` when called from the API.